apps/mobile/src/features/swap/swap-state/swap-state.utils.ts

```diff
@@ -0,0 +1,136 @@
+import { SecondaryAmount, SwapInternalState } from '@/features/swap/swap-state/swap-state.types';
+import { whenInputCurrencyMode } from '@/utils/when-currency-input-mode';
+import BigNumber from 'bignumber.js';
+import { filter, pipe, sortBy } from 'remeda';
+
+import {
+  FungibleCryptoAsset,
+  Money,
+  isBtcAsset,
+  isSip10Asset,
+  isStxAsset,
+} from '@leather.io/models';
+import { AccountSwapAsset } from '@leather.io/services';
+import { getAssetId } from '@leather.io/utils';
+
+export function convertMoneyToInputValue(money: Money | null): string {
+  if (!money) return '';
+  return money.amount
+    .shiftedBy(-money.decimals)
+    .toFixed(money.decimals, BigNumber.ROUND_HALF_UP)
+    .replace(/\.0+$/, '')
+    .replace(/(\.\d*?)0+$/, '$1');
+}
+
+export function areSameAssets(first: FungibleCryptoAsset, second: FungibleCryptoAsset) {
+  const firstAssetId = getAssetId(first);
+  const secondAssetId = getAssetId(second);
+  return firstAssetId.protocol === secondAssetId.protocol && firstAssetId.id === secondAssetId.id;
+}
+
+export function adjustAmountForDecimals(amount: string, maxDecimals: number): string {
+  if (!amount || amount === '') return amount;
```

Comment:
nit: empty string is a falsy value, so it is going to be caught by `!amount` check anyway
```suggestion
  if (!amount) return amount;
```

Reply:
I blame the frenchman.